sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveComparisonTest.scala

```diff
@@ -480,7 +480,7 @@ abstract class HiveComparisonTest
                 val executions = queryList.map(new TestHive.QueryExecution(_))
                 executions.foreach(_.toRdd)
                 val tablesGenerated = queryList.zip(executions).flatMap {
-                  case (q, e) => e.sparkPlan.collect {
+                  case (q, e) => e.executedPlan.collect {
```

Comment:
We do extra processing for `sparkPlan` in `prepareForExecution`. It includes four rules now:

    override val batches: Seq[Batch] = Seq(
      Batch("Subquery", Once, PlanSubqueries(SessionState.this)),
      Batch("Add exchange", Once, EnsureRequirements(conf)),
      Batch("Whole stage codegen", Once, CollapseCodegenStages(conf)),
      Batch("Reuse duplicated exchanges", Once, ReuseExchange(conf))
    )

The error shown in PR description is happened because `sparkPlan` lacks the processing of `EnsureRequirements` (i.e., "Add exchange" rule). We will add additional node in `sparkPlan` to enable exchanging.